wpcom-legacy-redirector.php

```diff
@@ -86,7 +87,7 @@ static function insert_legacy_redirect( $from_url, $redirect_to ) {
 
 		$from_url_hash = self::get_url_hash( $from_url );
 
-		if ( false !== self::get_redirect_uri( $from_url ) ) {
+		if ( false !== self::get_redirect_uri( $from_url ) && ( 'add' === $mode ) ) {
```

Comment:
Maybe -- the lookup below is not strict (ie -- doesn't care about trailing slashes) whereas this check has so far been strict about trailing slashes.  Separate enhancement?